catalog/spatial/registry/registry-api-impl/src/test/java/org/codice/ddf/registry/api/impl/RegistryStoreCleanupHandlerTest.java

```diff
@@ -0,0 +1,236 @@
+import org.codice.ddf.registry.federationadmin.service.internal.FederationAdminService;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventConstants;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+
+public class RegistryStoreCleanupHandlerTest {
+    @Mock
+    private ExecutorService executorService;
+
+    @Mock
+    private FederationAdminService federationAdmin;
+
+    @Mock
+    private BundleContext context;
+
+    @Mock
+    private RegistryStore store;
+
+    private MetacardImpl mcard;
+
+    private Event event;
+
+    private Dictionary<String, Object> eventProperties;
+
+    private RegistryStoreCleanupHandler rsch;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        rsch = new RegistryStoreCleanupHandler() {
+            @Override
+            BundleContext getBundleContext() {
+                return context;
+            }
+        };
+        rsch.setExecutor(executorService);
+        rsch.setFederationAdminService(federationAdmin);
+        setupSerialExecutor();
+
+        when(context.getService(any(ServiceReference.class))).thenReturn(store);
+        when(store.getRegistryId()).thenReturn("registryId");
+
+        mcard = new MetacardImpl();
+        mcard.setId("id");
+        mcard.setModifiedDate(new Date());
+        mcard.setAttribute(Metacard.TAGS, RegistryConstants.REGISTRY_TAG_INTERNAL);
+        mcard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID, "registryId"));
+        eventProperties = new Hashtable<>();
+
+    }
+
+    @Test
+    public void testBindRegistryStoreNullReference() throws Exception {
+        ServiceReference ref = getServiceReference("servicePid");
+        handleEvent(null,
+                ref,
+                ServiceEvent.UNREGISTERING,
+                "registryId",
+                Collections.singletonList(mcard));
+        verify(federationAdmin,
+                never()).deleteRegistryEntriesByMetacardIds(Collections.singletonList(mcard.getId()));
+    }
+
+    @Test
+    public void testUnBindRegistryStore() throws Exception {
+        ServiceReference ref = getServiceReference("servicePid");
+        handleEvent(ref,
+                ref,
+                ServiceEvent.UNREGISTERING,
+                "registryId",
+                Collections.singletonList(mcard));
+        rsch.unbindRegistryStore(ref);
+        handleEvent(null,
+                ref,
+                ServiceEvent.UNREGISTERING,
+                "registryId",
+                Collections.singletonList(mcard));
+        verify(federationAdmin,
```

Comment:
Ok, that makes more sense. I think we can change it to just do a bindRegistryStore followed by an unbindRegistryStore. The first handleEvent is just does that for you (along with some other stuff.